iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -445,9 +444,9 @@ public static List<DeleteFile> getDeleteFiles(Table table, String partitionPath)
         t -> ((PositionDeletesScanTask) t).file()));
   }
 
-  public static Expression generateExpressionFromPartitionSpec(Table table, Map<String, String> partitionSpec)
-      throws SemanticException {
-    Map<String, PartitionField> partitionFieldMap = getPartitionFields(table).stream()
+  public static Expression generateExpressionFromPartitionSpec(Table table, Map<String, String> partitionSpec,
+      boolean latestSpecOnly) throws SemanticException {
+    Map<String, PartitionField> partitionFieldMap = getPartitionFields(table, latestSpecOnly).stream()
```

Comment:
The method `generateExpressionFromPartitionSpec` is called from two places and the actual parameter value `latestSpecOnly` is always `true`. How about calling
```
icebergTable.spec().fields()
```
instead of
```
getPartitionFields(table, latestSpecOnly)
```

Reply:
it's called from `IcebergTableUtil.getPartitionNames(Table icebergTable, Map<String, String> partitionSpec, boolean latestSpecOnly)`, that is called from `.HiveIcebergStorageHandlergetPartitions(org.apache.hadoop.hive.ql.metadata.Table table,
      Map<String, String> partitionSpec, boolean latestSpecOnly)`

Reply:
or are you saying that we could refactor it to
````
  public static List<PartitionField> getPartitionFields(Table table, boolean latestSpecOnly) {
    return latestSpecOnly ? table.spec().fields() :
      table.specs().values().stream()
        .flatMap(spec -> spec.fields().stream()).distinct()
        .collect(Collectors.toList());
  }
````

Reply:
> or are you saying that we could refactor it to
> 
> ```
>   public static List<PartitionField> getPartitionFields(Table table, boolean latestSpecOnly) {
>     return latestSpecOnly ? table.spec().fields() :
>       table.specs().values().stream()
>         .flatMap(spec -> spec.fields().stream()).distinct()
>         .collect(Collectors.toList());
>   }
> ```

yes.

Actually at the top of the call stack, from we call this method, we already know whether we need only the current spec or all. :)

Reply:
done